src/main/obsRecorder.ts

```diff
@@ -174,44 +177,132 @@ const setOBSVideoResolution = (res: Size, paramString: string) => {
-  const scene = osn.SceneFactory.create('test-scene');
+    default:
+      throw Error(`[OBS] Invalid capture mode: ${options.obsCaptureMode}`);
+  }
+
+  setOBSVideoResolution(baseResolution, 'Base');
+
+  const scene: IScene = osn.SceneFactory.create('test-scene');
   const sceneItem = scene.add(videoSource);
   sceneItem.scale = { x: 1.0, y: 1.0 };
 
+  console.log(`[OBS] Configured video input source with mode '${options.obsCaptureMode}'`, inspectObject(videoSource.settings))
+
+  watchVideoSourceSize(sceneItem, videoSource, baseResolution);
+
   return scene;
 }
 
+/**
+ * Create and return a game capture video source ('game_capture')
+ */
+const createGameCaptureSource = (): IInput => {
+  const videoSource = osn.InputFactory.create('game_capture', 'Game Capture');
+  const settings = videoSource.settings;
+
+  settings['capture_cursor'] = true;
+  settings['capture_mode'] = 'window';
+  settings['allow_transparency'] = true;
+  settings['priority'] = 1; // Window title must match
+  settings['window'] = 'World of Warcraft:GxWindowClass:Wow.exe';
+
+  videoSource.update(settings);
+  videoSource.save();
+
+  return videoSource;
+};
+
+/**
+ * Create and return a monitor capture video source ('monitor_capture')
+ */
+const createMonitorCaptureSource = (monitorIndex: number): IInput => {
+  const videoSource = osn.InputFactory.create('monitor_capture', 'Monitor Capture');
+  const settings = videoSource.settings;
+
+  settings['monitor'] = monitorIndex;
+
+  videoSource.update(settings);
+  videoSource.save();
+
+  return videoSource;
+};
+
+/**
+ * Watch the video input source for size changes and adjust the scene item
+ * scaling accordingly.
+ *
+ * @param sceneItem       Scene item as returned from `IScene.add()`
+ * @param sourceName      Video input source
+ * @param baseResolution  Resolution used as base for scaling the video source
+ */
+const watchVideoSourceSize = (sceneItem: ISceneItem, videoSource: IInput, baseResolution: Size): void => {
+  clearInterval(timerVideoSourceSize);
+  timerVideoSourceSize = setInterval(() => {
+    const result = { width: videoSource.width, height: videoSource.height };
+
+    if (result.width === 0 || result.height === 0) {
+      return;
+    }
+
+    if (lastVideoSourceSize && result.width === lastVideoSourceSize.width && result.height === lastVideoSourceSize.height) {
+      return;
+    }
+
+    lastVideoSourceSize = result;
+
+    const scaleFactor = baseResolution.width / result.width;
+    sceneItem.scale = { x: scaleFactor, y: scaleFactor };
+
+    const logDetails = {
+      base: baseResolution,
+      input: result,
+      scale: sceneItem.scale,
+    };
+
+    console.log("[OBS] Adjusting scene item scale due to video input source size change", inspectObject(logDetails));
+  }, 5000);
+};
+
 /*
 * setupSources
 */
 const setupSources = (scene: any, audioInputDeviceId: string, audioOutputDeviceId: string ) => {
+  clearSources();
```

Comment:
Not clear to me what the need for this change is?

Reply:
Might be spill over from audio sources PR ?

Reply:
No, not a spill over - intentionally ported over. We need to clear the output sources before we reconfigure them in order to properly release and remove them. It has worked before without it because we didn't change the video source. It's basically just cleaning up.